interview-questions.md

```diff
@@ -3,44 +3,46 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. In a model called Animal that has_many Sightings, what is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
 
-  Your answer:
+  Your answer: the foreign key would be part of the sightings model, and the foreign key would be called animal_id.
 
-  Researched answer:
+  Researched answer: naming conventions for a foreign key the target table name, followed by the source table name. The table containing the foreign key is called the child table, and the table containing the candidate key is called the referenced or parent table.
 
 
 
 2. Which routes must always be passed params and why?
 
-  Your answer:
+  Your answer: the routes that are creating, updating, destroying and showing all need to have params passed through them. This is so the user knows exactly what they are accessing and are creating.
 
-  Researched answer:
+  Researched answer: Parameters are how your Rails server receives information from the user to dynamically modify the view.
 
 
 
 3. Write a rails route for a controller called "main" and a page called "game" that takes in a parameter called "guess".
 
-  Your answer:
+  Your answer:  //@main = params.require(:game).permit(:guess)
 
 
 
 4. Name three rails generator commands. What is created by each?
 
-  Your answer:
+  Your answer:rails s - starts the server
+              rails new my_app -d postgresql -T - generates the rails application
+              rails g resource model my_model - generates and database table for app
 
-  Researched answer:
+  Researched answer: it's actually - rails g resource model blankets softness:string - both the migration and model that I need to start using guitars as an ActiveRecord class. It created a controller (and a guitar views folder) so that I can start coding up some controller actions. And finally, it plugged in resourced routes for my guitars.
```

Comment:
good research! here is the complete list:

rails g migration - creates a migration to change the database 
rails g model - create a new table with all the table attributes 
rails g controller - create a new controller for a specific model 
rails g resource - create the routes, model, controller